### ISSUE TITLE
bower2nix includes ~ in generated versions

### ISSUE THREAD

**Issue by Baughn** [rvl/bower2nix#3]
E.g. for a bower.json such as this:

```
{
  "name": "jupyter-notebook-deps",
  "version": "0.0.1",
  "dependencies": {
    "backbone": "components/backbone#~1.2",
    "bootstrap": "components/bootstrap#~3.3",
    (snip)
  }
}
```

...the generated .nix file will include ~ in the version number:

```
 (fetchbower "backbone" "components/backbone#~1.2" "components/backbone#~1.2" "1arcv99907bzr84lfrm9qyv0l8al653p6m4ih188n2dz8dd7n6ls")
 (fetchbower "bootstrap" "components/bootstrap#~3.3" "components/bootstrap#~3.3" "1vf8wylj51dh3gkjasnifpycvsfaxjci9hj4vd5q028ck7fprk8g")
```

Not only does this break bowerVersion, but it would also make the fetch non-fixed in the event that bowerVersion were fixed.

You do handle ~-ranges correctly in other cases; at the moment, I suspect it's the hash that's throwing it off.

**Comment by rvl** [comment]
It is indeed the hash which is throwing it off. I'm not even sure what `#~1.2` is supposed to mean. In this case I suggest tweaking the generated nix until a solution can be found.

**Comment by Baughn** [comment]
It seems to mean.. "Look for GitHub releases matching this version, approximately; install that."

**Comment by johbo** [comment]
It seems to work if we use the source specifier and the resolved version, e.g. `components/bootstrap#3.3`. Without the source specifier (e.g. just `3.3`) I ran into trouble because it looked in the wrong place.

**Comment by johbo** [comment]
I've added my workaround in https://github.com/rvl/bower2nix/pull/6

**Comment by rvl** [comment]
Thanks @johbo, workarounds are good. I will check & merge later.

**Comment by rvl** [comment]
Fixed by 16521d4f502145f6bcaaff992aa13e251f39a930, sorry for the delay.
